File: Haskell/beginner/2003.hs
-- https://judge.beecrowd.com/pt/problems/view/2003
-- Domingo de Manhã

-- Imports
import System.IO (isEOF)


-- Main
main :: IO ()
main = loop
  where
    loop = do
      eof <- isEOF
      if eof
        then return ()
        else do
          line <- getLine
          let [hourStr, minuteStr] = splitOn ':' line
              hour = read hourStr :: Int
              minute = read minuteStr :: Int
              
          putStr "Atraso maximo: "
          if hour < 7 || (hour == 7 && minute == 0)
            then putStrLn "0"
            else putStrLn . show $ (hour - 7) * 60 + minute
          loop

-- Helper to Split
splitOn :: Char -> String -> [String]
splitOn delimiter = foldr (\c acc -> if c == delimiter then "" : acc else (c : head acc) : tail acc) [""]

-- By: Fernando Serra
-- https://github.com/fernandosserra

File: Python/beginner/2003.py
# https://judge.beecrowd.com/pt/problems/view/2003
# Domingo de Manhã

while True:
    try:
        hour, minute = map(int, input().split(':'))
        if hour < 7 or (hour == 7 and minute == 0):
            print("Atraso maximo: 0")
        elif hour == 7 and minute == 0:
            print("Atraso maximo: 0")
        else:
            delay = (hour - 7) * 60 + minute
            print(f"Atraso maximo: {delay}")
    except EOFError:
        break
    except ValueError:
        break
    
# By: Fernando Serra
# https://github.com/fernandosserra 